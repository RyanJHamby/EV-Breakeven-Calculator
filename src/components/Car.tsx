import React, { FC } from "react";

export interface CarProps {
  // Define your props interface here
  Make_ID: number;
  Make_Name: string;
  Model_ID: number;
  Model_Name: string;
}

export const Car: FC<CarProps> = (props): JSX.Element => {
    console.log("hello")
    return (
        <div>
            <h3>{props.Model_ID}</h3>
            <p>{props.Model_Name}</p>
            <p>{props.Make_ID}</p>
            <p>{props.Make_Name}</p>
        </div>
    )
}