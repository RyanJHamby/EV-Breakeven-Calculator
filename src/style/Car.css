.mainPageCar {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 1rem;
    padding: 1rem;
    border: 1px solid black;
    border-radius: 1rem;
    width: 18vw;
    height: 18vw;
    background: white;
}

.carModelTitle {
    font-size: 1.5rem;
    font-weight: bold;
    margin: 0;
    padding: 0;
    text-align: center;
}

.carThumbnail {
    width: 90%;
    /* height: 50%; */
    object-fit: contain;
}

.carShortInfoSection {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 90%;
}

.carShortInfoSectionItem {
    font-size: .8rem;
}