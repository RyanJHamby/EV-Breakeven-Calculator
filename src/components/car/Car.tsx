import React, { FC } from "react";
import "../../style/Car.css";

export type CarProps = {
    id: number;
    model: string;
    photo_url: string;
    manufacturer_name: string;
    model_year: number;
    fuel_name: string;
    drivetrain: string;
    electric_range: string;
}

export const Car: FC<{ obj: CarProps }> = (props): JSX.Element => {
    const capitalizeWords = (input: string) => {
        if (input) {
            const processedWords = input.split(" ").map((word) => {
                return word.charAt(0).toUpperCase() + word.slice(1);
            })
            return processedWords.join(" ");
        }
        return "";
    }

    const processMileage = (input: string) => {
        if (input) {
            return input.includes("-") || Number(input) > 70 ? input + " mi range" : input + " MPG";
        } else {
            return "Unknown MPG"
        }
    }

    return (
        <div className="mainPageCar">
            <h3 className="carModelTitle">{capitalizeWords(props.obj.manufacturer_name) + " " + capitalizeWords(props.obj.model)}</h3>
            <div className="carShortInfoSection">
                <div className="carShortInfoSectionItem">{props.obj.model_year}</div>
                <div className="carShortInfoSectionItem">{capitalizeWords(props.obj.fuel_name)}</div>
                <div className="carShortInfoSectionItem">{processMileage(props.obj.electric_range)}</div>
            </div>
            <img src={props.obj.photo_url ? props.obj.photo_url : "unknownCar.jpeg"} alt="car" className="carThumbnail"/>
        </div>
    )
}