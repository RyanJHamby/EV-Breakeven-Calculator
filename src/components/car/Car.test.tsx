import React from 'react'
import renderer from 'react-test-renderer';
import { Car } from "./Car"

it('changes the class when hovered', () => {
    const component = renderer.create(
        <Car 
            obj={{
                "id": 1,
                "model": "test-model",
                "photo_url": "test-url",
                "manufacturer_name": "test-manufacturer",
                "model_year": 2023,
                "fuel_name": "test-fuel",
                "drivetrain": "test-drivetrain",
                "electric_range": "test-range"
            }}
        />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});