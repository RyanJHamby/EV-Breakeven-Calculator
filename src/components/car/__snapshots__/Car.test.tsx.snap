// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders car component 1`] = `
<div
  className="mainPageCar"
>
  <h3
    className="carModelTitle"
  >
    Test-manufacturer Test-model
  </h3>
  <div
    className="carShortInfoSection"
  >
    <div
      className="carShortInfoSectionItem"
    >
      2023
    </div>
    <div
      className="carShortInfoSectionItem"
    >
      Test-fuel
    </div>
    <div
      className="carShortInfoSectionItem"
    >
      test-range mi range
    </div>
  </div>
  <img
    alt="car"
    className="carThumbnail"
    src="test-url"
  />
</div>
`;
